compiler/frontend/src/org/jetbrains/kotlin/diagnostics/PositioningStrategies.kt

```diff
@@ -745,19 +743,29 @@ object PositioningStrategies {
         }
     }
 
-    val REFERENCE_BY_QUALIFIED: PositioningStrategy<PsiElement> = object : PositioningStrategy<PsiElement>() {
+    val REFERENCE_BY_QUALIFIED: PositioningStrategy<PsiElement> = FindReferencePositioningStrategy(false)
+    val REFERENCED_NAME_BY_QUALIFIED: PositioningStrategy<PsiElement> = FindReferencePositioningStrategy(true)
+
+    class FindReferencePositioningStrategy(val locateReferencedName: Boolean) : PositioningStrategy<PsiElement>() {
         override fun mark(element: PsiElement): List<TextRange> {
-            when (element) {
+            var result = when (element) {
                 is KtQualifiedExpression -> {
                     when (val selectorExpression = element.selectorExpression) {
-                        is KtCallExpression -> return mark(selectorExpression.calleeExpression ?: selectorExpression)
-                        is KtReferenceExpression -> return mark(selectorExpression)
+                        is KtCallExpression -> selectorExpression.calleeExpression ?: selectorExpression
+                        is KtReferenceExpression -> selectorExpression
+                        else -> element
                     }
                 }
-                is KtCallExpression -> return mark(element.calleeExpression ?: element)
-                is KtConstructorDelegationCall -> return mark(element.calleeExpression ?: element)
+                is KtCallExpression -> element.calleeExpression ?: element
+                is KtConstructorDelegationCall -> element.calleeExpression ?: element
+                is KtOperationExpression -> element.operationReference
+                is KtWhenConditionInRange -> element.operationReference
+                else -> element
             }
-            return super.mark(element)
+            while (locateReferencedName && result is KtParenthesizedExpression) {
```

Comment:
Same here. It'd be better to see examples that explain the role of `KtParenthesizedExpression` in loop condition.

Reply:
OK, added example in the doc for `locateReferencedName` instead.